tests/driver_cst816s/main.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the CST816S touch screen driver
+ *
+ * @author      koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "cst816s_params.h"
+#include "cst816s.h"
+#include "xtimer.h"
+#include "fmt.h"
+#include "thread.h"
+#include "thread_flags.h"
+
+static kernel_pid_t _pid;
+
+#define CST816S_THREAD_FLAG     (1 << 8)
+#define CST816S_NUM_TOUCHES     5
+
+static void _cb(cst816s_t *dev, void *arg)
+{
+    (void)dev;
+    (void)arg;
+    thread_flags_set((thread_t *)sched_threads[_pid], CST816S_THREAD_FLAG);
```

Comment:
might just have used a mutex instead, but it's good to have an example for `thread_flags_set()` too

Reply:
I find using thread flags for waking up a thread simpler than wrapping a mutex around the thing.

Reply:
Not sure what you mean by wrapping, I'd just use `mutex_lock()` to wait for the interrupt and `mutex_unlock()` in the interrupt handler.

Reply:
But thread flags are slightly faster!

Anyway, do you want to continue this discussion or are we going to keep this around as a thread flag example? :stuck_out_tongue: